### ISSUE TITLE
Error when `COPY`ing `STRING[]` from Parquet file

### ISSUE THREAD

**Issue by linozen** [kuzudb/kuzu#2703]
When I run the following command, I get a segmentation fault:

```cypher
COPY Release FROM 'output/nodes_releases.parquet';
```

This is an excerpt from the coredump:
```
Core was generated by `kuzu_shell ./graph/'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000000000b35dce in kuzu::storage::VarListColumnChunk::append(kuzu::common::ValueVector*) ()
[Current thread is 1 (LWP 7231)]
```

[This](https://github.com/kuzudb/kuzu/blob/8bae60ddbf5ab4c3772c74a080143dac1587c08f/src/include/storage/store/var_list_column_chunk.h#L27) seems to be the relevant code.

When I remove the `STRING[]` property from the schema, everything works.

**Comment by acquamarin** [comment]
Hi Linus!

Do you mind sending me the dataset so i can reproduce the bug? To locate the bug quickly, can you run: `load from 'output/nodes_releases.parquet' return *` and see whether you still got a seg fault?

**Comment by linozen** [comment]
Found what `LOAD` does. Linking the docs here for reference: https://kuzudb.com/docusaurus/cypher/query-clauses/load_from

**Comment by acquamarin** [comment]
Hi Linus!
Thanks for helping me locating the bug. 
So `LOAD FROM` simply scans the parquet file, and returns what the reader has scanned.
`COPY XXX FROM` is:  scan parquet file and write what the reader has scanned to the storage.
Looks like the bug is inside the copy operator.
Thanks for reporting the debug. Our team will take a look soon!

**Comment by linozen** [comment]
Just FYI: The code where I use KuzuDB lives here: https://git.fsfe.org/TEDective/parser/src/branch/main/tedective_parser/graph.py

I now fixed all the segfaults I ran into by simple flattening the data, i.e. replacing structs and Lists simply by strings with a prefixed key. Maybe you could point users who run into segfaults in that direction. Just an idea though.

BTW: Importing the entire public procurement data from the EU since 2017 (10 million nodes, 30 million edges) took about about 24 seconds in KuzuDB. That's soo much faster than everything else I have tried, it does not even make sense to calculate the speed-up. 

Will now get into querying. Thanks for the amazing code.

**Comment by acquamarin** [comment]
Hi Linus!

I did a quick patch which should fix the query #2706 :
`COPY Release FROM 'output/nodes_releases.parquet';`

Can you pull the newest master and try whether it has fixed your problem?

**Comment by prrao87** [comment]
Yes, Kùzu's speed over existing solutions is one of its standout attributes. 😀

So if I understand correctly, the issue is that you have nested pydantic schemas that are part of a larger table? From what I gather, the columns that have the nested data structures aren't being used themselves as node/edge attributes in your case - they're just part of a larger table whose flat columns are used as node/edge attributes.

You're right that we could provide more useful error-handling that tells the user "If you have nested data, please use flattened tables and see if it resolves the issue.

If you experience any further issues with importing or querying, let us know. Closing this for now as it seems you have it resolved.
